source/guides/github-actions-ci-cd-sample/publish-to-test-pypi.yml

```
@@ -6,6 +6,8 @@ jobs:
   build-n-publish:
     name: Build and publish Python 🐍 distributions 📦 to PyPI and TestPyPI
     runs-on: ubuntu-latest
+    permissions:
```

It's strongly recommend to use environments to further lock it down:
```suggestion
    environment:
      name: pypi
      url: https://pypi.org/p/<package-name>
    permissions:
```